### DISCUSSION TITLE
How to add langs/locales list in nuxtjs-i18n configuration dynamically.

### DISCUSSION THREAD

**Discussion by seetpalsingh** [nuxt-modules/i18n#2794] [Q&A]
Hi Can someone plz guide me , how i can update/add list of locales in i18n configuration.

In nuxt.config.js i don't have list of langs. 
but on SSR I get list of locals/langs from strapi, that i want to add in i18 lang options list.

How i can do this.

**Comment by MevenGedivote** [comment]
Hello, I'm in a similar situation, I'm trying to find a way to dynamically set the list of locales with a list that I receive from a request to the backend. My app is not SSR though. 
We can configure in the backend of our app the list of languages that are available to the users, so that the app can be used by different clients in different configurations.

If anyone has a solution for this, it would be greatly appreciated 😊

 For now we've implemented a hack : we've been setting an extended list in the i18n configuration with all the possible languages that our app has ever been used with (10+ languages) and manually checking in the onBeforeLanguageSwitch hook that the language selected by the user is in the list of configured languages received from the backend. 
But it's getting hard to maintain, because this technical bit is not known by the peoples that configure the list of possible languages available to the user for each app version, and it leads to occasionnal bugs when they're trying to add a language that's missing from the list.
Plus, the route list gets bloated with a duplicate of every routes for every locale, even though the app is most of the time configured with 1-2 possible languages.
